pkg/icingadb/schema.go

```diff
@@ -44,13 +44,26 @@ func CheckSchema(ctx context.Context, db *database.DB) error {
 		return errors.Wrap(err, "can't check database schema version")
 	}
 
-	if version != expectedDbSchemaVersion {
+	if len(versions) == 0 {
+		return fmt.Errorf("no database schema version is stored in the database")
+	}
+
+	for i := 0; i < len(versions)-1; i++ {
```

Comment:
To be honest, I had to look twice to see how it works. So it might be a good idea to explain here how it works and that we rely on incremental, non-optional schema upgrade version numbers. (We might change that in the future and then you can see here immediately that there's something to be done).

Reply:
Good point. I have both added a paragraph to the function documentation and added a short explanation above the loop, explaining the pair-wise comparison to ensure no gaps were left.